src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -8413,617 +9036,195 @@ void SketchObject::rebuildExternalGeometry(bool defining, bool addIntersection)
         const std::string &SubElement=SubElements[i];
         const std::string &key = keys[i];
 
+        bool beingCreated = false;
+        if (extToAdd) {
+            beingCreated = extToAdd->obj == Obj && extToAdd->subname == SubElement;
+        }
+
+        bool projection = Types[i] == (int)ExtTypes::Projection || Types[i] == (int)ExtTypes::Both;
+        bool intersection = Types[i] == (int)ExtTypes::Intersection || Types[i] == (int)ExtTypes::Both;

```

Comment:
If you do my comment about flags you would only need to check if flag is present.